### PR TITLE
Update packet_login_plugin_response

### PR DESCRIPTION
**Pull request by frej4189**
This closes https://github.com/PrismarineJS/mineflayer/issues/2940

### CODE REVIEW

**Comment by nickelpro** [comment]
Disagrees with `wiki.vg`, someone have the decompiled netty code handy?

**Comment by nickelpro** [comment]
The current protocol is correct per the 1.19.3 source:

From `ServerboundCustomQueryPacket`
```java
public void write(FriendlyByteBuf friendlyByteBuf3) {
    friendlyByteBuf3.writeVarInt(this.transactionId);
    friendlyByteBuf3.writeNullable(this.data, (friendlyByteBuf, friendlyByteBuf2) -> friendlyByteBuf.writeBytes(friendlyByteBuf2.slice()));
}
```
A "nullable" is a boolean prefixed `restBuffer`
From `FriendlyByteBuf`
```java
public <T> void writeNullable(@Nullable T t, Writer<T> writer) {
    if (t != null) {
        this.writeBoolean(true);
        writer.accept(this, t);
    } else {
        this.writeBoolean(false);
    }
}
```

**Comment by frej4189** [comment]
> The current protocol is correct per the 1.19.3 source:
> 
> From `ServerboundCustomQueryPacket`
> ```java
> public void write(FriendlyByteBuf friendlyByteBuf3) {
>     friendlyByteBuf3.writeVarInt(this.transactionId);
>     friendlyByteBuf3.writeNullable(this.data, (friendlyByteBuf, friendlyByteBuf2) -> friendlyByteBuf.writeBytes(friendlyByteBuf2.slice()));
> }
> ```
> A "nullable" is a boolean prefix
> From `FriendlyByteBuf`
> ```java
> public <T> void writeNullable(@Nullable T t, Writer<T> writer) {
>     if (t != null) {
>         this.writeBoolean(true);
>         writer.accept(this, t);
>     } else {
>         this.writeBoolean(false);
>     }
> }
> ```

That's what the option type is for

**Comment by nickelpro** [comment]
I have no problem with you making the restBuffer an optional, but that's not what you did here

**Comment by nickelpro** [comment]
Oh RIP I see what happened
It got both, derp

**Comment by frej4189** [comment]
> I have no problem with you making the restBuffer an optional, but that's not what you did here

It already is optional. The boolean is an extra parameter in lower versions

**Review by nickelpro — APPROVED** [review]
@rom1504

File: data/pc/1.19.3/protocol.json [hunk 1]
```diff
@@ -2396,9 +2392,9 @@
         "packet_profileless_chat": [
         	"container",
           	[
-    		  {
```
Comment (extremeheat) [inline data/pc/1.19.3/protocol.json:2399]: bad tabulation

Comment (frej4189) [reply data/pc/1.19.3/protocol.json:2399]: I see. Don't know how that slipped in. Will update

Comment (frej4189) [reply data/pc/1.19.3/protocol.json:2399]: Fixed.

**Review by extremeheat — APPROVED** [review]

**Pull request merged**